src/Puzzle.java

```diff
@@ -181,6 +195,21 @@ public List<String> getFoundWords() {
     public int getEarnedPoints() {
         return this.earnedPoints;
     }
+
+    /**
+     * Gets the current rank achived for the puzzle.
+     * 
+     * @return the current rank achived for the puzzle
+     */
+    public Rank getRank() {
+        Rank currentRank = null;
+        for (Rank rank : Rank.values()) {
+            if (earnedPoints >= rank.getRequiredPoints(totalPoints)) {
+                currentRank = rank;
```

Comment:
Breaking the loop after the rank is assigned would save a few iterations.

Reply:
Will do.

Reply:
That will only ever return beginner. I could reverse the array but I think that would add more processing than just doing it this way.

Reply:
But every rank below the current rank also satisfies the condition.